geoopt/manifolds/poincare/__init__.py

```diff
@@ -45,58 +44,72 @@ def _check_point_on_manifold(self, x, atol=1e-5, rtol=1e-5):
             reason = None
         return ok, reason
 
-    def _check_vector_on_tangent(self, x, u, atol=1e-5, rtol=1e-5):
+    def _check_vector_on_tangent(self, x, u, *, atol=1e-5, rtol=1e-5):
         return True, None
 
-    def _dist(self, x, y, keepdim):
+    def _dist(self, x, y, *, keepdim=False):
```

Comment:
It's probably time to tackle `Manifold` itself and strip the underscore-names. Although, I don't see much trouble in merging in an intermediate state

Reply:
The trouble in moving `_method` to `method` is docstring. The docstring has to be copy pasted

Reply:
Sounds like least of the troubles. Anyway. it's evening but I'm not thinking straight anymore. What you think about merging current stuff right now and then expmapping from the new state?

Reply:
I would like to hear from @rrkarim first, and then merge if no more changes are proposed

Reply:
Also good to notify @bcol23 and @jdieter31 who open sourced ther research on top of geoopt. They might comment on changes as well

Reply:
Anyway I hope we merge in few days, ideally to hear from all of you today (I might be too optimistic)